internal/provider/kubernetes/predicates.go

```diff
@@ -294,7 +294,7 @@ func (r *gatewayAPIReconciler) validateServiceForReconcile(obj client.Object) bo
 	// Check if the Service belongs to a Gateway, if so, update the Gateway status.
 	gtw := r.findOwningGateway(ctx, labels)
 	if gtw != nil {
-		r.updateStatusForGateway(ctx, gtw)
+		r.updateGatewayStatus(gtw)
```

Comment:
This is a bit difficult for the later comer to understand and mantain the Gateway status update logic here, instead of firing up an empty GatewayStatus update event, the predicate should just return true and handle the GatewayStatus properly in the Reconcile method. This can be addressed in a follow-up PR.